MergerService/Utils/TaskUtils.cs

```diff
@@ -178,32 +46,74 @@ public TaskUtils(IConfigurationManager configuration, IHttpRequestUtils httpClie
+            string baseUrl = this._configuration.GetConfiguration("TASK", "overseerUrl");
+            string relativeUri = $"tasks/{jobId}/{taskId}/completed";
+            string url = new Uri(new Uri(baseUrl), relativeUri).ToString();
+            _ = this._httpClient.PostDataString(url, null, false);
+        }
 
-            // Convert metadata to json
-            var jsonSerializerSettings = new JsonSerializerSettings();
-            jsonSerializerSettings.Converters.Add(new StringEnumConverter());
-            string json = JsonConvert.SerializeObject(updateParameters, jsonSerializerSettings);
-            var body = new System.Net.Http.StringContent(json, Encoding.UTF8, "application/json");
+        private void Update(string jobId, string taskId, FormUrlEncodedContent content)
+        {
+            string baseUrl = this._configuration.GetConfiguration("TASK", "jobManagerUrl");
+            string relativeUri = $"jobs/{jobId}/tasks/{taskId}";
+            string url = new Uri(new Uri(baseUrl), relativeUri).ToString();
+            _ = this._httpClient.PutDataString(url, content, false);
+        }
 
-            _ = this._httpClient.PutDataString(url, body);
+        public void UpdateProgress(string jobId, string taskId, int progress)
+        {
+            var content = new FormUrlEncodedContent(new[]
+            {
+                new KeyValuePair<string, string>("percentage", progress.ToString())
+            });
+            Update(jobId, taskId, content);
         }
 
         public void UpdateCompletion(string jobId, string taskId)
         {
-            // Update overseer on task completion
-            string baseUrl = this._configuration.GetConfiguration("TASK", "overseerUrl");
-            string url = Path.Combine(baseUrl, $"tasks/{jobId}/{taskId}/completed");
-            _ = this._httpClient.PostDataString(url);
+            var content = new FormUrlEncodedContent(new[]
+            {
+                new KeyValuePair<string, string>("percentage", "100"),
+                new KeyValuePair<string, string>("status", "completed")
+            });
+            Update(jobId, taskId, content);
+        }
+
+        public void UpdateReject(string jobId, string taskId, int attempts, string reason, bool resettable)
+        {
+            attempts++;
+
+            // Check if the task should actually fail
+            if (!resettable || attempts == this._maxAttempts)
+            {
+                UpdateFailed(jobId, taskId, reason);
+            }
+
+            var content = new FormUrlEncodedContent(new[]
+            {
+                new KeyValuePair<string, string>("attempts", attempts.ToString()),
+                new KeyValuePair<string, string>("reason", reason),
+                new KeyValuePair<string, string>("resettable", resettable.ToString())
+            });
+            Update(jobId, taskId, content);
+        }
+
+        public void UpdateFailed(string jobId, string taskId, string reason)
```

Comment:
should be private

Reply:
No, should be public. It is called on exceptions when handling the task.